docs/build-results-archive.md

```diff
@@ -0,0 +1,75 @@
+# Build results archive.
+
+This document describes a build results archive format which can be used to store the sarif results of a build-and-test procedure.
+
+Once created this archive and its contents _should not_ be modified by any downstream program so that the integrity of the original build and its results can be maintained.
+
+Applications which integrate the data and enhance or elaborate it are _recommended_ to maintain their own data storage for these enhancements.
+
+## Filename and path conventions
+
+Archives generated locally _should_ make use of the `log` subdirectory generated by colcon.
+Within the log directory a `build_results_archives` directory _should_ be created if it does not already exist.
+colcon uses the format `{build,test}_YYYY-MM-DD_HH-MM-SS` however this format is not ISO8601 parse-able and is ambiguously in local time.
+Since one intended use case of build results archives is sharing and comparing between builds on different machines (thus potentially in different time zones) the timestamp format used for build result archive filenames is explictly in UTC
+The recommended filename is formatted `build_results_YYYYMMDDTHHMMSSZ.tar.bz2` where `YYYYMMDDTHHMMSSZ` is a basic-formatted ISO8601 UTC date and time.
+
+When a new build results archive is created locally, a symlink in the `log/build_results_archives` subdirectory with the name `latest_build_results.tar.bz2` _should_ be updated to link relatively to the just-created build archive.
+
+## Contents
+
+The archive will take the form of a bzip2 compressed tar file containing the files described below.
+
+```
+./build-results-archive
+./build/**/*.sarif
+./colcon-build-cmd
+./colcon-test-cmd
+./processed.sarif
+./timestamp
+./vcs-export-exact.repos
+```
+
+The individual files are further described:
+
+### build-results-archive
+
+This is a JSON file containing metadata regarding the current archive.
+The file _must_ contain an single top-level JSON object and that object _must_ contain the `"version"` key with the value of `1`.
+Future revisions of this format may add additional valid versions.
+
+### build/**/*.sarif
+
+The build/ subdirectory _must_ contain the unaltered sarif content of the workspace build directory after the end of the latest colcon test run.
+
+### colcon-build-cmd
+
+This file _must_ contain the full colcon build command described in the colcon logs under the `latest_build` symlink.
+The file, when run as a /bin/sh program or prefixed with the `exec` shell command in a workspace root _should_ invoke colcon in the local environment using the same arguments as the original build.
+
+### colcon-test-cmd
+
+This file is identical in behavior to `colcon-build-cmd` but _must_ describe the test command under the `latest_test` symlink.
+
+### processed-sarif-cmd
+
+This file is optional but recommended.
+If it is present, the processed.sarif file _must_ also be present.
```

Comment:
Is this left behind from a previous iteration? My understanding is we decided for multiple `.sarif` files (corroborated below in processed/**/*.sarif), but this might be for something else that I'm forgetting about.

Reply:
corrected in e7eded4